src/dlgPackageExporter.cpp

```diff
@@ -268,64 +317,230 @@ void dlgPackageExporter::slot_export_package()
-    if (err != 0) {
-        zip_error_to_str(buf, sizeof(buf), err, errno);
-        //FIXME: report error to user qDebug()<<"dp zip open error"<<zipFile<<buf;
-        close();
-        return;
-    }
-    QDir dir(tempDir);
-    QStringList contents = dir.entryList();
-    for (int i = 0; i < contents.size(); i++) {
-        QString fname = contents[i];
-        if (fname == "." || fname == "..") {
-            continue;
-        }
-        QString fullName = tempDir + "/" + contents[i];
-        struct zip_source* s = zip_source_file(archive, fullName.toStdString().c_str(), 0, 0);
-        if (s == nullptr) {
-            int sep = 0;
-            zip_error_get(archive, &err, &sep);
-            zip_error_to_str(buf, sizeof(buf), err, errno);
-            //FIXME: report error to userqDebug()<<"zip source error"<<fullName<<fname<<buf;
-        }
-        err = zip_file_add(archive, fname.toStdString().c_str(), s, ZIP_FL_OVERWRITE);
-        if (err == -1) {
-            int sep = 0;
-            zip_error_get(archive, &err, &sep);
-            zip_error_to_str(buf, sizeof(buf), err, errno);
-            //FIXME: report error to userqDebug()<<"added file error"<<fullName<<fname<<buf;
+    if (isOk) {
+        // zip error code:
+        int ze = 0;
+/*
+* Relevant options:
+* * ZIP_CREATE creates the archive if it does not exist.
+* * ZIP_TRUNCATE zaps any contents in a previously existing file.
+*/
+
+        // We previously only used ZIP_CREATE but without ZIP_TRUNCATE that
+        // will always break when we used zip_add() to replace config.lua
+        // because we should have used zip_replace() instead (and the 0.1x
+        // versions with zip_add() only lacks the option of the later
+        // zip_file_add() which DOES have an overwrite option in 1.x)
+        zip* archive = zip_open(mPackagePathFileName.toUtf8().constData(), ZIP_CREATE|ZIP_TRUNCATE, &ze);
+
+        if (!archive) {
+            // Failed to open/create archive file
+            // A better error handling system (not requiring a previously
+            // defined finite-sized char type buffer {which obviously can
+            // have string buffer overflow issues} is available in 1.x
+            // versions of libzip):
+            zip_error_t error;
+            zip_error_init_with_code(&error, ze);
+            displayResultMessage(tr("Failed to open module (archive) file, error is: \"%1\".")
+                                 .arg(QString::fromUtf8(zip_error_strerror(&error))), false);
+            zip_error_fini(&error);
+            isOk = false;
+            // The above flag will now cause execution to drop down to the bottom of
+            // the method and set up a means to close the dialogue after the user
+            // has seen the error message...
+
+        } else {
+            // Opened/created archive file successfully
+/*
+ * Previous code here failed if the user included sub-directories in the
```

Comment:
Not sure how commenting the older Mudlet versions will help in the future.
Maybe left over from designing and developing this change? Seems obsolete.

Reply:
The prototype I lifted this from supported (as far as possible) all `libzip` versions (including pre- 0.11) so I did have to remind myself (and anyone else interested) which things had to/could be done for each version (inside `#if LIBZIP_VERSION_MAJOR`/`MINOR <= ...` conditional compilation block...